do.py

```diff
@@ -13,18 +13,21 @@
 os.environ["PATH"] = os.environ["PATH"] + ":{0}/go/bin:{0}/bin".format(os.environ["GOPATH"])
 
 
+def arch():
+    return getattr(platform.uname(), "machine", platform.uname()[-1]).lower()
+
+
 def on_arm():
-    arch = platform.uname().machine.lower()
-    return arch == "arm64" or arch == "aarch64"
+    return arch() in ["arm64", "aarch64"]
 
 
 def on_x86():
-    arch = platform.uname().machine.lower()
-    return arch == "x86_64"
+    return arch() == "x86_64"
 
 
 def on_linux():
-    return sys.platform == "linux"
+    print("The platform is {}".format(sys.platform))
+    return "linux" in sys.platform
```

Comment:
use `==`

Reply:
I had ran into a problem when the platform name was linux2, that's when I changed to in

Reply:
ok got it